pandas/tests/indexes/test_base.py

```diff
@@ -445,14 +445,22 @@ def test_constructor_dtypes_timedelta(self, attr, klass):
         result = klass(list(values), dtype=dtype)
         tm.assert_index_equal(result, index)
 
-    def test_constructor_empty_gen(self):
+    def test_constructor_empty_list(self):
```

Comment:
Could you combine this test with the one below with `pytest.mark.paremeterize` over `skip_index_keys` and the empty list and generator?